ert3/console/_console.py

```diff
@@ -232,17 +239,65 @@ def _init(args: Any) -> None:
         ) from err
 
 
+def _build_local_testrun_config(
+    experiment_run_config: ert3.config.ExperimentRunConfig,
+) -> ert3.config.ExperimentRunConfig:
+    if experiment_run_config.experiment_config.type != "evaluation":
+        sys.exit(
+            "Local test runs are only supported for evaluation experiments, was:\n"
+            f"{experiment_run_config.experiment_config.type}"
+        )
+
+    for input_record in experiment_run_config.ensemble_config.input:
+        if not (
+            input_record.source.startswith("stochastic.")
+            or input_record.source.startswith("resources.")
+        ):
+            sys.exit(
```

Comment:
Think we should raise an more specific ErtError here (and the same above), and let the calling code handle the exiting. Possibly a new error type is needed. That would also allow for easier testing of these special cases, which would be nice to have included :slightly_smiling_face:

Reply:
Since this relates to #3016 I suggest NotImplementedError.